### ISSUE TITLE
Add support for transforming OBJs with texture

### ISSUE THREAD

**Issue by saeedzr** [mikedh/trimesh#105]
We have OBJs with lines starting with `v`, `vn`, `vt` and `f`. As of now, the exported OBJ from `trimesh` only exports `v` and `f` lines. That results in mesh texture info being lost. Would be great to add support for that.

**Comment by mikedh** [comment]
Hey thanks for the PR the other day! I think in order to fully support textures, there would need to be a `trimesh.visual.TextureVisuals` object or something. Implementing that would be quite a bit of work however. As a temporary hack, I added your texture line to mesh.metadata['vertex_texture'] and will add the corresponding line to the exporter at some point.

Also, do you have a small OBJ model that has texture that could be added to the tests?

Thanks!
Mike

**Comment by saeedzr** [comment]
Thanks for the response Mike. 

Yeah I realized later that my PR was not really complete and correct and didn't solve the issue in a proper way. Needed more work.

I ended up creating my own parser and transformer and added it to our stack. Some points that may wanna know for applying transformation to Wavefront OBJs (if you don't know already):
- `v` lines should be transformed with the result of `tmat4x4` * `x y z [w=1]`. There might also be extra `r b g [a]` values at the end and they should stay untouched (This is already working I guess, at least partly)
- `vn` lines should be transformed with the result of `rmat3x3` * `i j k`
- `vt` lines should not be touched, just relay to export.
- `f` lines should not be touched. And for textures to work properly, they need to be in `x/x/x y/y/y z/z/z`
- There are references to material files in the source OBJ, they also need to relayed to export.

Here is a link to a small OBJ with textures:
https://storage.googleapis.com/imerso-3dscanner/saeed%40imerso.com%2F85a0b195-0768-4432-a38b-575bc282d02b%2F6e963678-8994-4384-87b6-4c9e8966197b%2Fobj-6e963678-8994-4384-87b6-4c9e8966197b.zip?response-content-disposition=attachment%3B+filename%3D20170602_103456_obj.zip&Expires=2216394185&GoogleAccessId=scan-worker%40imerso-3dscanner-backend.iam.gserviceaccount.com&Signature=xhx%2BBwj4W7M1p0Be1S6DrmAw%2F8B3vmUo42L%2BBJ4zphb43xqDVB%2FCtqZKTY9JV4hFyzrPojpxHpS2o1gN0U%2FXTLevDYdiNLALj1Qp%2FWB0dNdaJ%2FwI1%2F9%2BbT3nMA%2BAZOgglU1OayrGoGvOT4FSJKKE3VEROk%2BahqNR3Rn3ejXxfysNXR9w9caiPeDdTNP%2B%2Fn4BLp4jX%2FbGLNTmucbeXXCnKGTq2RkMDEz6ImZYPaeJ5MIk53Z4lUi9zUHZN5Eo%2F%2FskbfXgoIyBg%2BddemokLjXMFkY0VOZAytQiDLxe2sDF8VCiyywLF%2F8JY0lkTXKjr7H6i2tPg2HcwbWPfR34htinwQ%3D%3D

Appreciate the good work!
Saeed
